internal/controller/buildpipeline/buildpipeline_adapter.go

```diff
@@ -305,3 +266,68 @@ func (a *Adapter) updateBuildPipelineRunWithFinalInfo(canRemoveFinalizer bool) e
 		return nil
 	})
 }
+
+// checkForSnapshotsCount makes sure that only one snapshot is associated with build pipelinerun and annotate that PLR with snapshot
+// informs about fact when PLR is associated with more existing snapshots
+func (a *Adapter) checkForSnapshotsCount(canRemoveFinalizer *bool, existingSnapshots *[]applicationapiv1alpha1.Snapshot) (result controller.OperationResult, err error) {
```

Comment:
function is called `check` but check functions doesn't update resources, semantically name is wrong.

it should be called `ensureBuildPLRSWithSnapshotAnnotation`

Reply:
Agreed, updated.